pkg/controller/atlasproject/custom_roles_test.go

```diff
@@ -1,210 +1,194 @@
 package atlasproject
 
 import (
+	"context"
+	"net/http"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
-	"go.uber.org/zap"
+	"github.com/stretchr/testify/mock"
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"go.mongodb.org/atlas-sdk/v20231115008/mockadmin"
```

Comment:
@roothorp do you think we should go already with the 2024 version here?
Or it might be better to complete the migration on the same one everywhere and then sweep upgrade in one or a few PRs.
cc @s-urbaniak

Reply:
i suggest to sync this with the DRI @helderjs

Reply:
Latest version contains significant changes and migration requires a separate PR